File: app/src/main/java/cn/surine/schedulex/ui/login/LoginViewModel.java
package cn.surine.schedulex.ui.login;

import android.text.Editable;
import android.text.TextUtils;
import android.text.TextWatcher;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import cn.surine.schedulex.base.Constants;
import cn.surine.schedulex.base.http.BaseHttpSubscriber;
import cn.surine.schedulex.base.utils.SimpleTextWatcher;
import cn.surine.schedulex.base.utils.Strs;
import cn.surine.schedulex.data.entity.VmResultString;

public class LoginViewModel extends ViewModel {

    private LoginRepository mLoginRepository;

    private MutableLiveData<String> account = new MutableLiveData<>();
    private MutableLiveData<String> password = new MutableLiveData<>();


    /**
     * 登录状态
     */
    public MutableLiveData<Integer> loginStatus = new MutableLiveData<>();
    public static final int START_LOGIN = 0;
    public static final int LOGIN_SUCCESS = 1;
    public static final int LOGIN_FAIL = 2;


    public TextWatcher accountWatcher = new SimpleTextWatcher() {
        @Override
        public void afterTextChanged(Editable s) {
            account.setValue(s.toString());
        }
    };
    public TextWatcher passwordWatcher = new SimpleTextWatcher() {
        @Override
        public void afterTextChanged(Editable s) {
            password.setValue(s.toString());
        }
    };


    public LoginViewModel(LoginRepository mLoginRepository) {
        this.mLoginRepository = mLoginRepository;
    }


    /**
     * 登录操作
     */
    public void login() {
        if(TextUtils.isEmpty(account.getValue()) || TextUtils.isEmpty(password.getValue())){
            return;
        }
        loginStatus.setValue(START_LOGIN);
        mLoginRepository.login(account.getValue(), password.getValue()).subscribe(new BaseHttpSubscriber<VmResultString>() {
            @Override
            public void onSuccess(MutableLiveData<VmResultString> vm) {
                //TODO:这边如果getValue为空的话，可能会出问题
                loginStatus.setValue(Strs.equals(Constants.LOGIN_SUCCESS, vm.getValue().result) ? LOGIN_SUCCESS : LOGIN_FAIL);
            }
        });
    }


}

File: app/src/main/java/cn/surine/schedulex/ui/login/LoginViewModel.kt
package cn.surine.schedulex.ui.login

import android.text.Editable
import android.text.TextUtils
import androidx.lifecycle.MutableLiveData
import cn.surine.schedulex.base.Constants
import cn.surine.schedulex.base.controller.BaseViewModel
import cn.surine.schedulex.base.http.BaseHttpSubscriber
import cn.surine.schedulex.base.utils.SimpleTextWatcher
import cn.surine.schedulex.data.entity.VmResultString

/**
 * Intro：
 *
 * @author sunliwei
 * @date 2020/6/26 11:18
 */
class LoginViewModel(val mLoginRepository: LoginRepository) : BaseViewModel() {

    private val account = MutableLiveData<String>()
    private val password = MutableLiveData<String>()


    /**
     * 登录状态
     */
    var loginStatus = MutableLiveData<Int>()


    companion object {
        val START_LOGIN = 0
        val LOGIN_SUCCESS = 1
        val LOGIN_FAIL = 2
    }


    var accountWatcher = object : SimpleTextWatcher() {
        override fun afterTextChanged(s: Editable?) {
            super.afterTextChanged(s)
            account.value = s.toString()
        }
    }
    var passwordWatcher = object : SimpleTextWatcher() {
        override fun afterTextChanged(s: Editable?) {
            super.afterTextChanged(s)
            password.value = s.toString()
        }
    }


    /**
     * 登录操作
     */
    fun login() {
        if (TextUtils.isEmpty(account.value) || TextUtils.isEmpty(password.value)) {
            return
        }
        loginStatus.value = START_LOGIN
        mLoginRepository.login(account.value!!, password.value!!).subscribe(object : BaseHttpSubscriber<VmResultString?>() {
            override fun onSuccess(vm: MutableLiveData<VmResultString?>?) {
                loginStatus.value = if (Constants.LOGIN_SUCCESS == vm?.value!!.result) LOGIN_SUCCESS else LOGIN_FAIL
            }
        })
    }
}